model_v4.py

```diff
@@ -0,0 +1,382 @@
+                        getattr(self, f'inception_{addition_name}_3x3_bn1')))
+
+            setattr(self, f'inception_{addition_name}_3x3_conv2',
+                          Conv2D(filters=conv2_3x3,
+                                 kernel_size=(3, 3),
+                                 strides=strides,
+                                 padding='same',
+
+                                 kernel_regularizer=l2(0.01),
+                                 name=f'inception_{addition_name}_3x3_conv2')
+                          (getattr(self, f'inception_{addition_name}_3x3_act1')))
+            setattr(self, f'inception_{addition_name}_3x3_bn2',
+                    BatchNormalization(name=f'inception_{addition_name}_3x3_bn2')
+                    (getattr(self, f'inception_{addition_name}_3x3_conv2')))
+            setattr(self, f'inception_{addition_name}_3x3_act2',
+                    Activation('relu', name=f'inception_{addition_name}_3x3_act2')(
+                        getattr(self, f'inception_{addition_name}_3x3_bn2')))
+            layers_set_list.append(getattr(self, f'inception_{addition_name}_3x3_act2'))
+
+        if addition_pool:
+            mp = MaxPooling2D(pool_size=(3, 3),
+                              strides=(2, 2),
+                              padding='same')(input_layer)
+            layers_set_list.append(mp)
+
+        out_layer = Concatenate(axis=3)(layers_set_list)
+        return out_layer
+
+    def make_prediction(self, data):
+        """
+        Data encodeing
+        :param data: list with images data
+        :return: list with encoded images - predictions
+        """
+        predictions = self.model.predict(data)
+        return predictions
+
+    def data_encoding(self, data):
+        """
+        Encoding of data
+        :param data: dict {name of image: image data}
+        :return: dict {name of image: encoded data}
+        """
+        data_for_prediction = np.array(list(data.values()))
+        predictions = self.make_prediction(data_for_prediction)
+        encoded_dict = {list(data.keys())[idx]: predictions[idx] for idx in range(len(predictions))}
+        return encoded_dict
+
+    def model_evaluation(self, data, threshold=0.7):
+        """
+        Make predictions for data and creation of dict with likelyhood vectors
+        :param data: numpy array (N, H, W, C)
+        :param threshold: float number with max of distance between vectors
+        :return: dict {idx of image: set of names}
+        """
+        predictions = self.make_prediction(data)
+        dist_mtx = pairwise_distances(predictions)
+        mask = dist_mtx <= threshold
+        result_dict = {idx: set(np.argwhere(mask[idx]).reshape(-1)) ^ {idx} for idx in range(mask.shape[0])}
+        return result_dict
+
+    def create_model_plot(self):
+        """
+        Create plot with model graph
+        :return: None
+        """
+        plot_model(self.model, to_file='model.png', show_shapes=True, show_layer_names=True)
```

Comment:
It will force me to setup _pydot_ package to my virtual environment. Could you please make it optional with turned off by default ?